params/precompile_config_test.go

```diff
@@ -57,30 +55,147 @@ func TestValidateWithChainConfig(t *testing.T) {
 			TxAllowListConfig: precompile.NewTxAllowListConfig(big.NewInt(5), admins),
 		},
 	)
-	err = badConfig.VerifyPrecompileUpgrades()
+	err = badConfig.Verify()
 	assert.ErrorContains(t, err, "disable should be [true]")
 }
 
-func TestValidate(t *testing.T) {
+func TestVerifyPrecompileUpgrades(t *testing.T) {
 	admins := []common.Address{{1}}
-	config := &ChainConfig{}
-	config.PrecompileUpgrades = []PrecompileUpgrade{
+	tests := []struct {
+		name          string
+		upgrades      []PrecompileUpgrade
+		expectedError string
+	}{
+		{
+			name: "enable and disable tx allow list",
+			upgrades: []PrecompileUpgrade{
+				{
+					TxAllowListConfig: precompile.NewTxAllowListConfig(big.NewInt(1), admins),
+				},
+				{
+					TxAllowListConfig: precompile.NewDisableTxAllowListConfig(big.NewInt(2)),
+				},
+			},
+			expectedError: "",
+		},
 		{
-			TxAllowListConfig: precompile.NewTxAllowListConfig(big.NewInt(1), admins),
+			name: "invalid allow list config in tx allowlist",
+			upgrades: []PrecompileUpgrade{
+				{
+					TxAllowListConfig: precompile.NewTxAllowListConfig(big.NewInt(1), admins),
+				},
+				{
+					TxAllowListConfig: precompile.NewDisableTxAllowListConfig(big.NewInt(2)),
+				},
+				{
+					TxAllowListConfig: &precompile.TxAllowListConfig{
+						AllowListConfig:   precompile.AllowListConfig{AllowListAdmins: admins, EnabledAddresses: admins},
+						UpgradeableConfig: precompile.UpgradeableConfig{BlockTimestamp: big.NewInt(3)},
+					},
+				},
```

Comment:
can we use constructors to set up all of the precompile configs within the precompile config validation tests?